guardian/recipes-backend
# Announce new recipes to EventBridge

## What does this change?

Outputs contents of new recipes to the eventbus as well as takedown notifications for recipes.

## How to test

- Create an SQS queue for testing
- Add a Rule to the `crier-eventbus-content-api-crier-v2-CODE` event bus with the following subscription, and link it to your SQS queue:
```json
{
  "source": ["recipe-responder"]
}
```
- (these two steps have been done already, going to the queue `test-recipes`)
- Go to the SQS console and open your queue
- Publish a recipe in CODE, using Composer (publishing only to the Feast channel will work too 😁 )
- In the SQS console, poll for new messages
- You should see messages coming through, indicating that a recipe was updated
- Perform a takedown on your recipe (or channel takedown)
- You should see another message indicating that the recipe was removed

## How can we measure success?

Able to update the search index automatically

## Have we considered potential risks?

n/a.  We have discussed potential issues from clustering a lot of messages on one bus, but concluded that since the buses are stateless and we are not throwing around millions of messages per second it should be fine.
